### PR TITLE
lease: fix racey access to 'leaseRevokeRate'

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/8193.

### CODE REVIEW

File: lease/lessor_test.go [hunk 1]
```diff
@@ -213,16 +214,19 @@ func TestLessorRenew(t *testing.T) {
 // TestLessorRenewExtendPileup ensures Lessor extends leases on promotion if too many
 // expire at the same time.
 func TestLessorRenewExtendPileup(t *testing.T) {
-	oldRevokeRate := leaseRevokeRate
-	defer func() { leaseRevokeRate = oldRevokeRate }()
-	leaseRevokeRate = 10
```
Comment (xiang90) [inline lease/lessor_test.go]: why there can be a lessor routine running when the test code changes leaseRevokeRate? we start the lessor at line 227 after this line. will there be concurrent tests running?

Comment (gyuho) [inline lease/lessor_test.go]: The race happens from the `runLoop` that keeps reading 'leaseRevokeRate`. And according to the stack trace, `TestLessorGrant ` was running concurrently.

**Comment by xiang90** [comment]
@gyuho the lessor in grant is not stopped. that is the problem: https://github.com/gyuho/etcd/blob/0645c514450983ce5284bd8305f310ef2be8fde2/lease/lessor_test.go#L45.

we should add a defer func to stop the lessor when test is finished.

**Comment by gyuho** [comment]
@xiang90 PTAL.

**Comment by xiang90** [comment]
lgtm

**Pull request merged**